src/Symfony/Component/Uid/DsHashablePolyfill.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uid;
+
+interface DsHashablePolyfill
+{
+    public function equals(mixed $other): bool;
+
+    public function hash(): mixed;
+}
+
+if (!interface_exists(\Ds\Hashable::class)) {
+    class_alias(DsHashablePolyfill::class, \Ds\Hashable::class);
```

Comment:
Declaring the interface here does not look right to me. This will potentially break application that check for the presence of the interface to detect if the extension is available.

Reply:
yeah true, didn't think of that, the current approach would be probably much better in that regard